### ISSUE TITLE
Never succeeded on using --dirver generic option to create docker machine

### ISSUE THREAD

**Issue by eyakcn** [docker/machine#3486]
I never succeed on using generic driver to create docker-machine before, and I had another try yesterday but unfortunately still failed...

Client:
OS: Windows 7
docker-machine: 0.7.0
Open SSH: 7.2

Host:
OS: RHEL 7.2
docker-engine: 1.11.1

Before I run docker-machine command, I made sure that id_rsa.pub was added to authorized_keys.
And when I run `ssh root@10.1.1.79` I can succeed login without password.

Then I run the docker-machine command, I got the error message which I completely can not understand...

C:\Users\songpeng>docker-machine -D create -d generic --generic-ip-address 10.1.1.79 --generic-ssh-key .ssh\id_rsa ci
Docker Machine Version:  0.7.0, build a650a40
Found binary path at D:\bin\docker-machine.exe
Launching plugin server for driver generic
Plugin server listening at address 127.0.0.1:50334
() Calling .GetVersion
Using API Version  1
() Calling .SetConfigRaw
() Calling .GetMachineName
(flag-lookup) Calling .GetMachineName
(flag-lookup) Calling .DriverName
(flag-lookup) Calling .GetCreateFlags
Found binary path at D:\bin\docker-machine.exe
Launching plugin server for driver generic
Plugin server listening at address 127.0.0.1:50338
() Calling .GetVersion
Using API Version  1
() Calling .SetConfigRaw
() Calling .GetMachineName
(ci) Calling .GetMachineName
(ci) Calling .DriverName
(ci) Calling .GetCreateFlags
(ci) Calling .SetConfigFromFlags
Running pre-create checks...
(ci) Calling .PreCreateCheck
(ci) Calling .GetConfigRaw
Creating machine...
(ci) Calling .Create
(ci) Calling .GetConfigRaw
(ci) Calling .DriverName
(ci) Calling .DriverName
Waiting for machine to be running, this may take a few minutes...
(ci) Calling .GetState
Detecting operating system of created instance...
Waiting for SSH to be available...
Getting to WaitForSSH function...
(ci) Calling .GetSSHHostname
(ci) Importing SSH key...
(ci) Calling .GetSSHPort
(ci) DBG | IP: 10.1.1.79
(ci) Calling .GetSSHKeyPath
(ci) Calling .GetSSHKeyPath
(ci) Calling .GetSSHUsername
Using SSH client type: external
&{[-F /dev/null -o BatchMode=yes -o PasswordAuthentication=no -o StrictHostKeyCh
ecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=quiet -o ConnectionAttempt
s=3 -o ConnectTimeout=10 -o ControlMaster=no -o ControlPath=none root@10.1.1.79
-o IdentitiesOnly=yes -i C:\Users\songpeng\.docker\machine\machines\ci\id_rsa -p
 22] C:\Program Files\OpenSSH\bin\ssh.exe <nil>}
About to run SSH command:
exit 0
SSH cmd err, output: <nil>:
Detecting the provisioner...
(ci) Calling .GetSSHHostname
(ci) Calling .GetSSHPort
(ci) Calling .GetSSHKeyPath
(ci) Calling .GetSSHKeyPath
(ci) Calling .GetSSHUsername
Using SSH client type: external
&{[-F /dev/null -o BatchMode=yes -o PasswordAuthentication=no -o StrictHostKeyCh
ecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=quiet -o ConnectionAttempt
s=3 -o ConnectTimeout=10 -o ControlMaster=no -o ControlPath=none root@10.1.1.79
-o IdentitiesOnly=yes -i C:\Users\songpeng\.docker\machine\machines\ci\id_rsa -p
 22] C:\Program Files\OpenSSH\bin\ssh.exe <nil>}
About to run SSH command:
cat /etc/os-release
SSH cmd err, output: <nil>: NAME="Red Hat Enterprise Linux Server"
VERSION="7.2 (Maipo)"
ID="rhel"
ID_LIKE="fedora"
VERSION_ID="7.2"
PRETTY_NAME="Red Hat Enterprise Linux Server 7.2 (Maipo)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:redhat:enterprise_linux:7.2:GA:server"
HOME_URL="https://www.redhat.com/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"

REDHAT_BUGZILLA_PRODUCT="Red Hat Enterprise Linux 7"
REDHAT_BUGZILLA_PRODUCT_VERSION=7.2
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="7.2"

Couldn't set key CPE_NAME, no corresponding struct field found
Couldn't set key , no corresponding struct field found
Couldn't set key REDHAT_BUGZILLA_PRODUCT, no corresponding struct field found
Couldn't set key REDHAT_BUGZILLA_PRODUCT_VERSION, no corresponding struct field
found
Couldn't set key REDHAT_SUPPORT_PRODUCT, no corresponding struct field found
Couldn't set key REDHAT_SUPPORT_PRODUCT_VERSION, no corresponding struct field f
ound
found compatible host: rhel
Provisioning with redhat...
No storagedriver specified, using devicemapper

(ci) Calling .GetMachineName
(ci) Calling .GetSSHHostname
(ci) Calling .GetSSHPort
(ci) Calling .GetSSHKeyPath
(ci) Calling .GetSSHKeyPath
(ci) Calling .GetSSHUsername
Using SSH client type: external
&{[-F /dev/null -o BatchMode=yes -o PasswordAuthentication=no -o StrictHostKeyCh
ecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=quiet -o ConnectionAttempt
s=3 -o ConnectTimeout=10 -o ControlMaster=no -o ControlPath=none root@10.1.1.79
-o IdentitiesOnly=yes -i C:\Users\songpeng\.docker\machine\machines\ci\id_rsa -p
 22] C:\Program Files\OpenSSH\bin\ssh.exe <nil>}
(ci) Calling .GetSSHHostname
(ci) Calling .GetSSHPort
(ci) Calling .GetSSHKeyPath
(ci) Calling .GetSSHKeyPath
(ci) Calling .GetSSHUsername
Using SSH client type: external
&{[-F /dev/null -o BatchMode=yes -o PasswordAuthentication=no -o StrictHostKeyCh
ecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=quiet -o ConnectionAttempt
s=3 -o ConnectTimeout=10 -o ControlMaster=no -o ControlPath=none root@10.1.1.79
-o IdentitiesOnly=yes -i C:\Users\songpeng\.docker\machine\machines\ci\id_rsa -p
 22] C:\Program Files\OpenSSH\bin\ssh.exe <nil>}
installing base package: name=curl
(ci) Calling .GetSSHHostname
(ci) Calling .GetSSHPort
(ci) Calling .GetSSHKeyPath
(ci) Calling .GetSSHKeyPath
(ci) Calling .GetSSHUsername
Using SSH client type: external
&{[-F /dev/null -o BatchMode=yes -o PasswordAuthentication=no -o StrictHostKeyCh
ecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=quiet -o ConnectionAttempt
s=3 -o ConnectTimeout=10 -o ControlMaster=no -o ControlPath=none root@10.1.1.79
-o IdentitiesOnly=yes -i C:\Users\songpeng\.docker\machine\machines\ci\id_rsa -p
 22] C:\Program Files\OpenSSH\bin\ssh.exe <nil>}
Error creating machine: Error running provisioning: exit status 1
open : The system cannot find the file specified.
notifying bugsnag: [Error creating machine: Error running provisioning: exit sta
tus 1]
bugsnag.Notify: [bugsnag/payload.deliver: Post https://notify.bugsnag.com/: dial
 tcp: lookup notify.bugsnag.com: getaddrinfow: This is usually a temporary error
 during hostname resolution and means that the local server did not receive a re
sponse from an authoritative server.]

**Comment by eyakcn** [comment]
I tried again, this time I run the client on RHEL 7.2, and the error message is the same.

I noticed there is such a message in the output:

installing base package: name=curl

Does this means the command try to install curl from a yum source?
If so, I guess it would always fail since there is no yum source configured currently

**Comment by eyakcn** [comment]
I configured a new yum source on the host, but still failed.

**Comment by JouniRajalaComiq** [comment]
Are you running behind corporate proxy? I had similar errors when I tried to provisioning through proxy server - didn't get it working with any options and had to revert to installing docker-engine with ansible before docker-machine create.

**Comment by eyakcn** [comment]
Yes, I run it inside company network. I also tried to install docker-engine before hand, but docker-machine create still do not work, with -D option I can not see detailed debug info... really sad
